include/seastar/core/reactor.hh

```diff
@@ -359,6 +360,7 @@ private:
     bool _have_aio_fsync = false;
     bool _kernel_page_cache = false;
     std::atomic<bool> _dying{false};
+    gate _background_gate;
 private:
```

Comment:
perhaps we could push this gate into a static thread_local member of manual_clock.

so manual_clock::advance() just needs to wrap its background task in with_gate(...).

and manual_clock::close() will wait on the gate, which is in turn called by reactor::close().

this way, we reduce the amount of changes to reactor.

I am not against adding this facility if it will be needed elsewhere. But perhaps we should consider the alternative I recommended if we're adding it only to handle a fragility in implementation detail of manual_clock.

Reply:
why limit the solution only to manual clock?
any other mechanism that would something alike will need to implement its own gate?
what's the point?

Reply:
My point above is that we're very conservative on changes to reactor. If we really need it, ok.

Reply:
We have other background tasks like destroying of foreign ptrs that deserve this protection. It's a matter of fortunate timing that they finish running before the reactor exits.

Reply:
See https://github.com/scylladb/seastar/blob/1eea95e0201eb676f860fcf3b4faf4f03666b217/include/seastar/core/sharded.hh#L844-L859